### PR TITLE
improve components structure

### PR DESCRIPTION
**Pull request by tilucasoli**
## Describe your changes
- More components available on `WidgetBook`
- Create a way to structure components
- Improve RemixSwitch UI

## Type of change
- **New feature** (non-breaking change which adds functionality)

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/components/checkbox/checkbox.style.dart [hunk 1]
```diff
@@ -4,31 +4,30 @@ import 'package:remix_ui/components/checkbox/checkbox.variants.dart';
 
 class CheckboxStyles extends StyleRecipe<CheckboxStyles> {
   const CheckboxStyles({
-    required this.outerContainer,
+    required this.flexContainer,
```
Comment (leoafarias) [inline lib/components/checkbox/checkbox.style.dart:7]: I believe `outerContainer` is a better description of the box itself. I do understand the problem you are trying to resolve.

File: lib/components/radio/radio.style.dart [hunk 2]
```diff
@@ -16,14 +16,13 @@ class RadioStyles extends StyleRecipe<RadioStyles> {
   final Style label;
   final Style row;
 
-  factory RadioStyles.build(
-      [RadioStyles? other, List<Variant> variants = const []]) {
+  factory RadioStyles.baseForm() {
```
Comment (leoafarias) [inline lib/components/radio/radio.style.dart:19]: Lets use `base` instead of `baseForm`

File: lib/components/button/button.style.dart [hunk 3]
```diff
@@ -13,9 +13,9 @@ class ButtonStyles extends StyleRecipe<ButtonStyles> {
   final Style icon;
   final Style label;
 
-  factory ButtonStyles.defaults() {
+  factory ButtonStyles.baseForm() {
```
Comment (leoafarias) [inline lib/components/button/button.style.dart:16]: Lets use `base` instead of `baseForm`

File: lib/components/switch/switch.style.dart [hunk 4]
```diff
@@ -52,22 +49,25 @@ class SwitchStyles extends StyleRecipe<SwitchStyles> {
 
 Style _outerContainerStyle() => Style(
       box.width(32),
-      box.height(20),
-      padding.all(2),
-      box.borderRadius.all(10),
+      box.height(16),
+      border.all.width(2),
```
Comment (leoafarias) [inline lib/components/switch/switch.style.dart]: WIth the latest beta on mix you can do `border.width()` and `border.strokeAlign()` so you don't need to call `all` method

**Review by leoafarias — CHANGES_REQUESTED** [review]

File: lib/components/radio/radio.style.dart (hunk 2 above)
Comment (tilucasoli) [reply lib/components/radio/radio.style.dart:19]: I guessed `base` was a keyword, but when I tested it just now, it turned out not to be

File: lib/components/switch/switch.style.dart (hunk 4 above)
Comment (tilucasoli) [reply lib/components/switch/switch.style.dart]: perfect

File: lib/components/checkbox/checkbox.style.dart (hunk 1 above)
Comment (tilucasoli) [reply lib/components/checkbox/checkbox.style.dart:7]: I renamed `outerContainer` to `flexContainer` to clarify that this style will be applied to a `FlexBox`.

File: lib/components/radio/radio.style.dart (hunk 2 above)
Comment (tilucasoli) [reply lib/components/radio/radio.style.dart:19]: Actually `base` is a keyword https://dart.dev/language/class-modifiers#base
https://dart.dev/language/keywords

File: lib/components/checkbox/checkbox.style.dart (hunk 1 above)
Comment (leoafarias) [reply lib/components/checkbox/checkbox.style.dart:7]: We should rename it to outerFlexContainer, so we know also the position, but let's discuss how to create a Style type that can indicate that is a flex style.

**Review by leoafarias — APPROVED** [review]

**Pull request merged**